infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InSnapshotSyncState.java

```diff
@@ -208,13 +202,6 @@ public void setTransitionEventId(UUID eventId) {
     private void cancelSnapshotSync(String cancelCause) {
         snapshotSender.stop();
         snapshotSender.getDataSenderBufferManager().getPendingMessages().clear();
-        if (!transmitFuture.isDone()) {
```

Comment:
If any event requires an ongoing snapshot sync to be cancelled, with the new change, it will be processed only after we have transitioned out of InSnapshotSyncState, right?

For example - request for a new snapshot sync state will be handled only after transmit() from the current snapshot sync has finished..

Reply:
Yes, that behaviour is the same as before...
Its not that all the transmit() method possible in snapshot sync has to be completed, its just that the current ongoing one has to be complete.
Before, since there were 2 different thread pools that were consuming the event and processing the event, this if condition was needed to have some kind of sync between the 2 threads. Now with only 1 thread processing an event for a given session, this kind of sync is not needed